nodejs/week2/app.js

```diff
@@ -0,0 +1,115 @@
+const express = require("express");
+const document = require("./document.json");
+const app = express();
+const router = express.Router();
+const port = process.env.PORT || 4000;
+
+
+// Support parsing JSON requests
+app.use(express.json());
+app.use('/search', router);
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+
+
+    
+router.get('/', async (req, res) => {
+  try {
+    const searchQuery = req.query.q;
+    if (searchQuery){
+
+      const searchResult = await document.filter(i => Object.values(i).some(val => (typeof val === 'string' || typeof val === "number") && val.toString().toLocaleLowerCase().includes(searchQuery.toLocaleLowerCase())));
+      if (searchResult.length){
+        res.send(searchResult)
+        return;
+      }
+      res.status(404)
+      throw new Error('Not Found')
+    }
+    res.send(document);
+  }
+  catch (error) {
+    throw error;
```

Comment:
I am not sure it makes sense to catch an error and then throwing it immediately after.
You could consider logging it to the console.

Reply:
Thanks for review, I have made some changes.